README
rbbr README
===========

  rbbr (RuBy BRowser) is an application to browse
  modules/classes hierarchy and their constants and methods. If
  the optionally modules to refer documents are installed, you
  can also refer the document of methods in the browser.

Requirements
------------

  * ruby 1.6.x, 1.8.x
  * Ruby/GTK2 0.7.0 or later
  * (optional) Ruby-GetText-Package 0.5.2 or later for L10n menus/labels.
               http://ponx.s5.xrea.com/hiki/ruby-gettext.html
               Also you need Racc-1.4.3 or later to install this package.
               http://www.ruby-lang.org/raa/list.rhtml?name=racc
  * (optional) Ruby-GNOME2 API Documents in RD+.
               http://ruby-gnome2.sourceforge.jp/data/ruby-gnome2-api.tar.gz
  * (optional) ri 0.8a (or ruby-1.8.0 or later)
               http://raa.ruby-lang.org/list.rhtml?name=ri
  * (optional) ReFe 0.7.1 or later (Japanese only)
               http://www.loveruby.net/en/refe.html

Install
-------
1. Install Racc, Ruby-GetText-Package, ri and ReFe. (All of them are optional)
2. De-Compress ruby-gnome2-api.tar.gz and rbbr's archive(Optional)
   withapi package include this api files.
3. Puts ruby-gnome2-api/* to (rbbr's top directory)/data/rbbr/rd/*
4. Move rbbr's top directory and type:

    $ ruby install.rb config
    $ ruby install.rb setup
   ($ su)
    # ruby install.rb install

  You can also install files in your favor directory by
  supplying install.rb some options. Try "ruby install.rb --help".


Gets Ruby-GNOME2 API References
-------------------------------
You can get the newest(nightly) archives from:

 http://ruby-gnome2.sourceforge.jp/data/ruby-gnome2-api.tar.gz

And you can overwrite them to old data which installed in 
(shared-data-dir)/rbbr/rd/.


Usage
-----

  * Linux, *BSD, Cygwin
    Run the following command:
    $ rbbr

  * MS Windows(except Cygwin)
    (Double or Single)Click ruby-install-dir\bin\rbbr.rbw from Explorer.

License
-------

  This program is free software.
  You can distribute/modify this program under
  the terms of the Ruby Distribute License. See COPYING.

Author
------
  (Ruby-GNOME2 and later) Ruby-GNOME2 Project Team <http://ruby-gnome2.sourceforge.jp>
  (Original Author)Hiroshi Igarashi <iga@ruby-lang.org>


ChangeLog
2004-03-28  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/methodlist.rb: Improved how to show arities.
	* lib/rbbr/ui/gtk/undo.rb: Fix to access class variables.
	This is the change specification of ruby-1.9.0. 
	Reported by Joao Pedrosa.
	* lib/rbbr/doc/rd.rb: Fix not to match to [],[]=.
	* lib/rbbr/doc/ri2.rb: Added.
	* po/ko/rbbr.po: Added. Translated by Park Ji-In.
	* AUTHORS: Add translators.

2004-03-27  Masao Mutoh  <mutoh@highway.ne.jp>

	* README, README.ja: Added Racc information. 
	* bin/rbbr: Convert it from dos-format to unix-format.
	Pointed out by Jan Weil.
	* lib/rbbr/ui/gtk/libselectiondialog.rb, po/*/rbbr.po: 
	Fixed a message. Pointed out by Jan Weil.
	* po/pt_BR/rbbr.po: Added. Translated by Joao Pedorosa.
	* AUTHORS: Add translators.

2004-03-26  Masao Mutoh  <mutoh@highway.ne.jp>

	* AUTHORS: Add translators.
	* po/de/rbbr.po: Added. Translated by Jan Weil.
	* po/cy/rbbr.po: Added. Translated by Dafydd Harries.
	* po/es/rbbr.po: Added. Translated by David Espada.
	* po/it/rbbr.po: Added. Translated by Gabriele Renzi.
	* po/fr/rbbr.po: Updated by Laurent Sansonetti.
	* README, README.ja: Revised.
	* lib/rbbr.rb: Update version information.
	* lib/rbbr/ui/gtk/searchentrybox.rb: Revise some strings.
	* po/*: Updated.
	* data/rbbr/icon.png: Updated.
	* lib/rbbr/ui/gtk/docview.rb, aboutdialog.rb: Reduce font
	warning under Win32.
	* pre-setup.rb: Work compiling catalog files under Win32.

2004-02-29  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/stockbrowser.rb: Remove GC.start.

2004-02-27  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/docviewer.rb: Fix a change of Ruby/GTK(Gdk::Window#pointer).

2004-02-24  Masao Mutoh  <mutoh@highway.ne.jp>

	* COPYING, COPYING.ja: Added.

2004-01-29  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/browselist.rb: Show blurb correctly under client's locale.
	* lib/rbbr/ui/gtk/moduledisplay.rb, browselist.rb: Add Child Properties tab.

2004-01-25  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/moduledisplay.rb, browelist.rb: 
	Support Styles for Gtk::Widgets. 

2004-01-16  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/browser.rb, moduledisplay.rb: 
	Add some keyboard accelerations.

2004-01-15  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/stockbrowser.rb, browser.rb: StockBrowser becomes 
	an modeless window.
	* lib/rbbr/ui/gtk/browser.rb: Add accelerator keys to List/Tree tabs.
	* lib/rbbr/ui/gtk/modulelabel.rb: Show texts larger.

2004-01-12  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/moduleview.rb: Improved scroll to search. 
	* lib/rbbr/ui/gtk/modulelabel.rb: Added hypertext function.
	* lib/rbbr/ui/gtk/moduledisplay.rb: Made link to Foo#bar work.

2004-01-11  Kenichi Komiya <kom@mail1.accsnet.ne.jp>

	* lib/rbbr/ui/gtk/moduledisplay.rb: Made link to class work.
	* lib/rbbr/ui/gtk/docviewer.rb: Fixed a bug which select all
	text when user follow a link.
	* lib/rbbr/ui/gtk/undo.rb: clear redo queue when new memnto
	is added.
	
2004-01-11  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/linkedtextbuffer.rb: Added hypertext function to docviewer.
	Original patch is produced by Kenichi Komiya.
	* lib/rbbr/ui/gtk/docviewer.rb: ditto.
	* lib/rbbr/ui/gtk/*.rb: Support undo/redo(Experimental).
	* lib/rbbr/ui/gtk/undo.rb: Added.
	* lib/rbbr/ui/gtk/searchentrybox.rb, browser.rb: Ctrl+T to move to the entrybox.

2004-01-04  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/modulelistview.rb, moduletreeview.rb, moduledisplay.rb
	methodlist.rb: Enable to search methods.
	* lib/rbbr/ui/gtk/searchentrybox.rb: Added(separate from browser.rb).
	* lib/rbbr/ui/gtk/moduleview.rb: Added as abstract class of 
	modulelistview.rb, moduletreeview.rb.

2004-01-03  Masao Mutoh  <mutoh@highway.ne.jp>

	* lib/rbbr/ui/gtk/stockbrowser.rb: Remove GC.start (Ruby/GTK2 was fixed in CVS).

2003-12-20  Laurent Sansonetti  <lrz@gnome.org>

	* lib/rbbr/ui/gtk.rb: Moved code to set the title of the
	main window in gtk/browser.rb.
	* lib/rbbr/ui/gtk/aboutdialog.rb: Removed the separator
	and fixed border width. 
	* lib/rbbr/ui/gtk/browser.rb: 
	- Only load scripts upon accept responses in the 'Open script' file
	  dialog ;
	- Display an error when a directory was selected instead
	  of a file in the 'Open script' file dialog ;
	- Added descriptive title for the error message boxes, when
	  failing to open script or compile regex ;
	- 'File/Require Library' accessible with CONTROL+L ;
	- 'File/Load Script' accessible with CONTROL+O ;
	- Replaced 'View/Update Information' with 'View/Refresh' ;
	- Removed an extra space in 'View/Stock Item and Icon Browser...' ;
	- Removed help icon for 'Help/About' ;
	- Translate the title of the main window ;
	- Disable 'View/[Expand|Collapse] all' when in list mode instead
	  of tree mode.
	* lib/rbbr/ui/gtk/libselectiondialog.rb:
	- Fixed the response of the Cancel button to allow closing of the
	  window by pressing Escape ;
	- Added descriptive text before the tree view ;
	- Removed the separator and fixed border width.
	* lib/rbbr/ui/gtk/moduledisplay.rb: Removed extra packing of tab
	labels (default packing is now used).
	* lib/rbbr/ui/gtk/stockbrowser.rb:
	- Automatically select the first item ;
	- Removed the separator and fixed border width.
	* lib/rbbr/ui/gtk/windowutils.rb: Fixed the error/warning dialog
	boxes to respect HIG alerts.

2003-12-27  Laurent Sansonetti  <lrz@gnome.org>

	* lib/rbbr/ui/gtk/docviewer.rb: Make the documentation viewer
	read-only while the 'hiki upload' feature isn't implemented.

2003-12-26  Laurent Sansonetti  <lrz@gnome.org>

	* lib/rbbr/ui/gtk.rb: Moved code to set the title of the
	main window in gtk/browser.rb.
	* lib/rbbr/ui/gtk/aboutdialog.rb: Removed the separator
	and fixed border width. 
	* lib/rbbr/ui/gtk/browser.rb: 
	- Only load scripts upon accept responses in the 'Open script' file
	  dialog ;
	- Display an error when a directory was selected instead
	  of a file in the 'Open script' file dialog ;

lib/rbbr/ui/gtk.rb
=begin

  rbbr/ui/gtk.rb - Meta-Level Information Browser User Interface with GTK+

  $Author: mutoh $
  $Date: 2004/03/26 13:04:14 $

  Copyright (C) 2002 Ruby-GNOME2 Project

  Copyright (C) 2000-2002 Hiroshi Igarashi <iga@ruby-lang.org>

  This program is free software.
  You can distribute/modify this program under
  the terms of the Ruby Distribute License.

=end

require 'observer'
require 'gtk2'
begin
  require 'gnome2'
  $GNOME_SUPPORTED = true
rescue LoadError
  $GNOME_SUPPORTED = false
end
require 'rbbr/config'
require 'rbbr/metainfo'
require 'rbbr/doc'
require 'rbbr/ui/gtk/undo'
require 'rbbr/ui/gtk/windowutils'
require 'rbbr/ui/gtk/browselist'
require 'rbbr/ui/gtk/methodlist'
require 'rbbr/ui/gtk/constlist'
require 'rbbr/ui/gtk/linkedtextbuffer'
require 'rbbr/ui/gtk/docviewer'
require 'rbbr/ui/gtk/moduleview'
require 'rbbr/ui/gtk/modulelistview'
require 'rbbr/ui/gtk/moduletreeview'
require 'rbbr/ui/gtk/browser'
require 'rbbr/ui/gtk/moduledisplay'
require 'rbbr/ui/gtk/modulelabel'
require 'rbbr/ui/gtk/stockbrowser'
require 'rbbr/ui/gtk/aboutdialog'
require 'rbbr/ui/gtk/libselectiondialog'
require 'rbbr/ui/gtk/confmanager'
require 'rbbr/ui/gtk/searchentrybox'

module RBBR
module UI
module GTK
  def self.main
    if $GNOME_SUPPORTED
      Gnome::Program.new("rbbr", RBBR::VERSION) 
    else
      Gtk.init
    end
    window = Browser.new
    window.show_all
    GLib::Log.set_handler("Gtk", 1|2|4|8) do |domain, level, message|
      # ignore log message
    end

	GC.start
    Gtk.main
  end

end
end
end

if $0 == __FILE__
  RBBR::UI::GTK.main
end
